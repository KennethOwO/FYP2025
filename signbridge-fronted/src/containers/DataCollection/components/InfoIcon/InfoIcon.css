.dataForm-info {
    position: relative;
    display: inline-block;
}

.dataForm-info svg {
    font-size: 40px;
    cursor: pointer;
}

.dataForm-info::before {
    content: attr(data-tooltip);
    position: absolute;
    bottom: 120%;
    left: 50%;
    transform: translateX(-50%);
    background-color: rgba(0, 0, 0, 0.501);
    color: white;
    padding: 4px 8px;
    border-radius: 4px;
    font-size: 14px;
    white-space: nowrap;
    opacity: 0;
    visibility: hidden;
    transition: opacity 0.2s;
    z-index: 1000;
}

.dataForm-info:hover::before {
    opacity: 1;
    visibility: visible;
}
