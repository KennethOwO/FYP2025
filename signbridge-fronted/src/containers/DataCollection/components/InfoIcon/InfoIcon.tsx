import React from "react";
import { BiSolidInfoSquare } from "react-icons/bi";
import { IconContext } from "react-icons";
import { useTranslation } from "react-i18next";
import "./InfoIcon.css";

interface InfoIconProps {
    onClick: () => void;
}

const InfoIcon: React.FC<InfoIconProps> = ({ onClick }) => {
    const { t } = useTranslation();

    return (
        <div className="dataForm-info" data-tooltip={t("instructions")}>
            <IconContext.Provider value={{ color: "#000" }}>
                <BiSolidInfoSquare onClick={onClick} />
            </IconContext.Provider>
        </div>
    );
};

export default InfoIcon;
