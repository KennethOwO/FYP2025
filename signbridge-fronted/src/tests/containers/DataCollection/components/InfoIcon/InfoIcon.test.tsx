import InfoIcon from "../../../../../containers/DataCollection/components/InfoIcon/InfoIcon";
import { create } from "react-test-renderer";
import { cleanup } from "@testing-library/react";
describe("Test InfoIcon", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const tree = create(<InfoIcon onClick={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
