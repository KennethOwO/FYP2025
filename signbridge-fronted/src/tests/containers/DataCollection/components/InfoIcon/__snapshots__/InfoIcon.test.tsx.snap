// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InfoIcon should render correctly 1`] = `
<svg
  className="dataForm-info"
  fill="currentColor"
  height="1em"
  onClick={[MockFunction]}
  stroke="currentColor"
  strokeWidth="0"
  style={
    {
      "color": "#000",
    }
  }
  viewBox="0 0 24 24"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M3 4v16a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1zm8 3h2v2h-2V7zm0 4h2v6h-2v-6z"
  />
</svg>
`;
